src/prf_expand.rs

```diff
@@ -108,3 +108,23 @@ impl PrfExpand<([u8; 96], [u8; 32], [u8; 4])> {
     pub const SAPLING_ZIP32_CHILD_NON_HARDENED: Self = Self::new(0x12);
 }
 with_inputs!(a, A, b, B, c, C);
+
+impl PrfExpand<([u8; 32], [u8; 4])> {
+    pub const REGISTERED_ZIP32_CHILD: Self = Self::new(0xAC);
+
+    /// Expands the given secret key in this domain, with additional `lead`
+    /// and `tag` inputs.
+    pub fn with_tag(
+        self,
+        c_par: &[u8],
+        sk_par: &[u8; 32],
+        i: &[u8; 4],
+        lead: Option<u8>,
+        tag: &[u8],
+    ) -> [u8; 64] {
+        match lead {
+            None => self.apply(c_par, &[sk_par, i, tag]),
+            Some(b) => self.apply(c_par, &[sk_par, i, &[b], tag]),
+        }
```

Comment:
https://github.com/zcash/zips/pull/978 says that the input to PRF^Expand is `sk_par || i` if `lead == 0 && tag.is_empty()`. Here if `lead = None` then the tag still gets appended. We should instead do something like `lead.unwrap_or(0)` and then the full check.